control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -1067,6 +1085,38 @@ func reconcileOauthServerRoute(routeInstance *routev1.Route) error {
 	return nil
 }
 
+func (r *HostedControlPlaneReconciler) reconcileOIDCRouteResources(ctx context.Context, hcp *hyperv1.HostedControlPlane) error {
+	route := manifests.OIDCRoute(hcp.GetNamespace())
+	r.Log.Info("Updating OIDC route")
+	_, err := controllerutil.CreateOrUpdate(ctx, r.Client, route, func() error {
+		rootCASecret := pki.RootCASecret(hcp.Namespace)
+		if err := r.Get(ctx, client.ObjectKeyFromObject(rootCASecret), rootCASecret); err != nil {
```

Comment:
No change requested here, but for future ref, the pattern has been to resolve reconcile dependencies (in this case `rootCASecret`) outside the reconcile function which make it easier to extract these into stateless functions and also avoids a pointless lookup of the target (`route`) if a dependency can't be satisfied